mule-extensions-api-persistence/src/main/java/org/mule/runtime/extension/api/persistence/ExtensionModelTypeAdapter.java

```diff
@@ -165,8 +166,10 @@ public ExtensionModel read(JsonReader in) throws IOException {
       }
 
       final ObjectType objectType = (ObjectType) type;
-      serializationContext.registerObjectType(objectType);
-      types.add(objectType);
+      if (isTypeRegistrable(objectType)) {
```

Comment:
you should not care about the type being registable or not. All you should care is if the type is present on ExtensionModel#getTypes() , otherwise you're coupling the serialization to the types business logic. Careful with this! Comparing with equals() might be fragile due to typeAnnotations being present or not. Maybe you should check by the TypeId